### ISSUE TITLE
Dandelion relaying for RPC calls.

### ISSUE THREAD

**Issue by srya-b** [gfanti/bitcoin#14]
RPC calls like `SendRawTransaction` should also attempts to use Dandelion propagation. Unlike transactions that we receive, `RelayTransactionDandelion` is not called in this instance. It appears that the function to modify in this case is `SendMessages`.

**Comment by srya-b** [comment]
Correction, `SendMessages` should retain its sole purpose of sending out messages. The functions like `sendrawtransaction` that already handle checks should take care of ensuring the correct nodes are sent the new transaction. Additionally, rpc functionality needs to have access to Dandelion structures and locks.

**Comment by amiller** [comment]
Good point, let's merge https://github.com/sbaks0820/bitcoin-dandelion/commit/621667e7c35d1f54e1ba6dbb32c0eaa599796a67 but with the new approach

**Comment by srya-b** [comment]
Yes, I'm going to go through the rpc calls and do the new inplementation.

**Comment by srya-b** [comment]
If we can include the namespace'd items and the new maps we've introduced like `mapEmbargo` in the `net_processing` header file, then we can copy the implementation in the rpc calls. However, there is a lot of stuff included in the anon namespace in `net_processing.cpp` and I don't know whether we want to expose all of those data structures to other parts of the code like rpc calls.

If that's the case maybe the best way would be just to create a new dandelion relaying function that can be called from rpc instead of doing the standard `pnode->pushInventory(...)`.

**Comment by amiller** [comment]
Right, I wanted to hide as much as possible of those data structures since they need to be locked and maintained consistently. What I did was expose the `RelayTransactionDandelion` function (it's declared in `net.h` right now, although to be fair that's a little confusing since its implemented in `net_processing.cpp`  https://github.com/gfanti/bitcoin/blob/66d0cf2643e25a60c33b7b990c4ddfc97a2a02b4/src/net.cpp#L2830), and used that in wallet.cpp https://github.com/gfanti/bitcoin/blob/dandelion/src/wallet/wallet.cpp#L1562
